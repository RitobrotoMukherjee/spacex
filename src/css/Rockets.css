.Rocket {
  display: flex;
  margin-bottom: 2rem;
}

.Rocket-Detail {
  display: flex;
  flex-direction: column;
}

.Rocket-Part1 {
  width: 30%;
  padding: 0 2rem;
}

.Rocket-Part2 {
  width: 70%;
  padding: 0 2rem;
}

.name {
  margin-bottom: 1rem;
}

.description {
  margin-bottom: 1rem;
  line-height: 1.5rem;
}

.rocket-container {
  margin-top: 10rem;
}

.btn-rocket {
  padding: 1rem;
  width: 10rem;
  border-color: black;
  border-radius: 10%;
}

.btn-cancel {
  background-color: white;
  color: black;
}

.btn-reserve {
  background-color: #295be4;
  color: white;
}

.reserved-state {
  color: white;
  background-color: #1bad40;
  padding: 0.3rem;
  width: fit-content;
  border-radius: 10%;
  margin-right: 0.3rem;
}
