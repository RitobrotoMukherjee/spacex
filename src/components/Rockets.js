import React, { useEffect } from 'react';
import '../css/Rockets.css';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';
import RocketsList from './rocketsComponents/RochetsList';

const Rockets = () => {
  const dispatch = useDispatch();
  const { loading, Rockets } = useSelector((state) => state.rockets);

  useEffect(() => {
    if (Rockets.length === 0) {
      dispatch(fetchRockets());
    }
  }, [dispatch]);

  return (

    <div>
      {loading && <div>Loading ...</div>}
      {!loading && (<RocketsList />)}

    </div>
  );
};

export default Rockets;
