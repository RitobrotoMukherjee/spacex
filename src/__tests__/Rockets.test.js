import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets';

it('Rockets component snapshot testing', () => {
  const rockets = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Rockets />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(rockets).toMatchSnapshot();
});
