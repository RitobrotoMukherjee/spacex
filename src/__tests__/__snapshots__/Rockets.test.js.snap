// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component snapshot testing 1`] = `
<div>
  <div />
</div>
`;
